metrics/scaling/k8s_scale_net.sh

```diff
@@ -0,0 +1,287 @@
+        local pods_json="$(cat << EOF
+	                "n_pods": {
+				"Result": ${n_pods},
+                                "Units" : "int",
+				"time_to_pod_net": ${net_time},
+				"Units" : "ms"
+                        }
+EOF
+        )"
+        metrics_json_add_array_fragment "$pods_json"
+
+	local launch_json="$(cat << EOF
+                        "launch_time": {
+                                "Result": $launch_time_ms,
+                                "Units" : "ms"
+                        }
+EOF
+        )"
+        metrics_json_add_array_fragment "$launch_json"
+
+	info "launch [$launch_time_ms]"
+
+	metrics_json_close_array_element
+}
+
+init() {
+        info "Initialising"
+
+	local cmds=("bc" "jq")
+	check_cmds "${cmds[@]}"
+
+	info "Checking Kubernetes accessible"
+	local worked=$( kubectl get nodes > /dev/null 2>&1 && echo $? || echo $? )
+	if [ "$worked" != 0 ]; then
+		die "kubectl failed to get nodes"
+	fi
+
+	info $(get_num_nodes) "Kubernetes nodes in 'Ready' state found"
+
+	k8s_api_init
+
+	 # And now we can set up our results storage then...
+	 metrics_json_init "k8s"
+	 save_config
+}
+
+save_config() {
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"testname": "${TEST_NAME}",
+                "NUM_DEPLOYMENTS": ${NUM_DEPLOYMENTS},
+                "STEP": ${STEP},
+                "wait_time": ${wait_time},
+                "delete_wait_time": ${delete_wait_time},
+                "settle_time": ${settle_time}
+        }
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+run() {
+	info "Running test"
+	local header_post="Content-Type: application/json"
+	local base_curl=${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments
+
+	trap cleanup EXIT QUIT KILL
+
+	metrics_json_start_array
+
+	for reqs in $(seq ${STEP} ${STEP} ${NUM_DEPLOYMENTS}); do
+		local deployment="${name_base_depl}${reqs}"
+		info "Testing replicas ${reqs} of ${NUM_DEPLOYMENTS}"
+		# Generate the next yaml file
+
+		local runtime_command
+		if [ -n "$RUNTIME" ]; then
+			runtime_command="s|@RUNTIMECLASS@|${RUNTIME}|g"
+		else
+			runtime_command="/@RUNTIMECLASS@/d"
+		fi
+
+		local input_template
+		local generated_file
+		if [ "$use_api" != "no" ]; then
+			input_template=$input_json
+			generated_file=$generated_json
+		else
+			input_template=$input_yaml
+			generated_file=$generated_yaml
+		fi
+
+		sed -e $runtime_command \
+			-e "s|@DEPLOYMENT@|${deployment}|g" \
+			-e "s|@LABEL@|${LABEL}|g" \
+			-e "s|@LABELVALUE@|${LABELVALUE}|g" \
+			-e "s|@GRACE@|${grace}|g" \
+			< ${input_template} > ${generated_file}
+
+		info "Applying changes"
+		local start_time=$(date +%s%N)
+
+		if [ "$use_api" != "no" ]; then
+			curl -s ${base_curl} -XPOST -H "${header_post}" -d@${generated_file} > /dev/null
+		else
+			kubectl apply -f ${generated_file}
+		fi
+
+		kubectl rollout status --timeout=${wait_time}s deployment/${deployment}
+		kubectl expose deployment $deployment
+
+		set +e
```

Comment:
urgh.... I know, trying to work around 'I know it might fail, but want to carry on but keep the result' can be hard with a `set -e` script... I had a horrid method I used over at https://github.com/kata-containers/tests/blob/master/lib/common.bash#L95

```bash
local is_valid=$( $rpath kata-env >/dev/null 2>&1 && echo $? || echo $? )
```

Consider if that might work here as well?

Reply:
Tested and did not work.
I'm keen to still using `set -e` as the other scaling tests does.